src/test/edu/oregonstate/cs361/battleship/BattleshipModelTest.java

```diff
@@ -57,7 +60,27 @@ public void playerShotTest() {
         assertTrue(theModel.getPlayerHits().size() == 0 && theModel.getPlayerMisses().size() == 0);
 
         theModel.updateShot("player", theModel.getComputerFireCoords());
-
         assertTrue(theModel.getPlayerHits().size() == 1 || theModel.getPlayerMisses().size() == 1);
+
+        assertTrue(theModel.updateShot("computer",new Coords(0,0)));
+        assertFalse(theModel.updateShot("computer",new Coords(1,1)));
+
+        theModel.updateShipPosition("computer","computer_aircraftCarrier",2,1,"vertical");
+        assertTrue(theModel.updateShot("computer", new Coords(1,2)));
+
+        assertTrue(theModel.updateShot("player",new Coords(0,0)));
+        assertFalse(theModel.updateShot("player",new Coords(1,1)));
+
+        theModel.updateShipPosition("player","aircraftCarrier",3,1,"vertical");
+        assertTrue(theModel.updateShot("player",new Coords(1,3)));
+        
+        //System.err.print() testing code from Stack-Overflow
+        ByteArrayOutputStream errorTest = new ByteArrayOutputStream();
+        System.setErr(new PrintStream(errorTest));
+
+        theModel.updateShot(null, new Coords(1,1));
+
+        assertEquals("Parameters not designated.", errorTest.toString());
```

Comment:
To me, this seems like a round-about way to test whether updateShot fails when the player/computer isn't specified (and null is passed in as targetSide), as you could just do updateShot and then see whether a shot actually gets placed. But it works, so I'm fine with it.